### PR TITLE
treefmt: enable actionlint, fix issues

### PR DESCRIPTION
**Pull request by katexochen**

### CODE REVIEW

File: .github/workflows/e2e_runtime-reproducibility.yml [hunk 1]
```diff
@@ -27,16 +27,17 @@ jobs:
       - uses: ./.github/actions/setup_nix
         with:
           githubToken: ${{ secrets.GITHUB_TOKEN }}
+          cachixToken: "" # Don't use the cachix cache
```
Comment (Freax13) [inline .github/workflows/e2e_runtime-reproducibility.yml:30]: This seems unrelated to enabling actionlint.

File: .github/workflows/e2e_runtime-reproducibility.yml [hunk 2]
```diff
@@ -27,16 +27,17 @@ jobs:
...
       - uses: nicknovitski/nix-develop@a2060d116a50b36dfab02280af558e73ab52427d # v1.1.0
       - name: Build
         id: build
         run: |
           nix build .#${{ matrix.build-target }} --option substituters https://cache.nixos.org --builders ""
-          reference_checksum="$(cat result/index.json | jq  -r '.manifests[0].digest')"
+          reference_checksum="$(jq  -r '.manifests[0].digest' result/index.json)"
```
Comment (Freax13) [inline .github/workflows/e2e_runtime-reproducibility.yml:36]: Nice

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -259,11 +259,7 @@ jobs:
           echo "initializerImgTagged=$(tag "$initializerImg")" | tee -a "$GITHUB_ENV"
           echo "serviceMeshImgTagged=$(tag "$serviceMeshImg")" | tee -a "$GITHUB_ENV"
       - name: Create file with image replacements
-        run: |
-          echo "ghcr.io/edgelesssys/contrast/coordinator:latest=$coordinatorImgTagged" > image-replacements.txt
-          echo "ghcr.io/edgelesssys/contrast/initializer:latest=$initializerImgTagged" >> image-replacements.txt
-          echo "ghcr.io/edgelesssys/contrast/service-mesh-proxy:latest=$serviceMeshImgTagged" >> image-replacements.txt
-          echo "ghcr.io/edgelesssys/contrast/node-installer-microsoft:latest=$nodeInstallerImgTagged" >> image-replacements.txt
+        run: "{\n  echo \"ghcr.io/edgelesssys/contrast/coordinator:latest=$coordinatorImgTagged\"\n  echo \"ghcr.io/edgelesssys/contrast/initializer:latest=$initializerImgTagged\"\n  echo \"ghcr.io/edgelesssys/contrast/service-mesh-proxy:latest=$serviceMeshImgTagged\"\n  echo \"ghcr.io/edgelesssys/contrast/node-installer-microsoft:latest=$nodeInstallerImgTagged\" \n} > image-replacements.txt\n"
```
Comment (Freax13) [inline .github/workflows/release.yml:262]: Yikes.

File: .github/workflows/e2e_runtime-reproducibility.yml (hunk 1 above)
Comment (katexochen) [reply .github/workflows/e2e_runtime-reproducibility.yml:30]: actionlint detects that this is a required input for the action.

**Review by Freax13 — APPROVED** [review]
